workspaces/marketplace/plugins/marketplace/src/components/InstalledPlugins/InstalledPluginsTable.tsx

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+
+import {
+  ResponseErrorPanel,
+  Table,
+  TableColumn,
+} from '@backstage/core-components';
+import { useApi } from '@backstage/core-plugin-api';
+
+import { Query, QueryResult } from '@material-table/core';
+
+import { DynamicPluginInfo, dynamicPluginsInfoApiRef } from '../../api';
+import { useInstalledPluginsCount } from '../../hooks/useInstalledPluginsCount';
+import EditIcon from '@mui/icons-material/Edit';
+import DeleteIcon from '@mui/icons-material/Delete';
+import FileDownloadOutlinedIcon from '@mui/icons-material/FileDownloadOutlined';
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import Tooltip from '@mui/material/Tooltip';
+
+import { processPluginsForDisplay } from '../../utils/pluginProcessing';
+
+export const InstalledPluginsTable = () => {
+  const [error, setError] = useState<Error | undefined>(undefined);
+  const [currentSearchTerm, setCurrentSearchTerm] = useState<string>('');
+  const { count } = useInstalledPluginsCount();
+  const dynamicPluginInfo = useApi(dynamicPluginsInfoApiRef);
+  let data: DynamicPluginInfo[] = [];
+  const columns: TableColumn<DynamicPluginInfo>[] = [
```

Comment:
This columns definition should be "static" (move it before the component) otherwise the installed plugins tab creates this warning:

```
The columns provided to material table are static, but contain functions which update on every render, resetting the table state. Provide a stable function or column reference or an row id to prevent state loss.
```